lib/matplotlib/collections.py

```diff
@@ -782,7 +782,8 @@ def update_scalarmappable(self):
         """Update colors from the scalar mappable array, if it is not None."""
         if self._A is None:
             return
-        if self._A.ndim > 1:
+        # QuadMesh can map 2d arrays
+        if self._A.ndim > 1 and not isinstance(self, QuadMesh):
```

Comment:
Could we do the raveling here as

```python
if isinstance(self, QuadMesh):
    self._A = self._A.ravel()
else:
    raise ValueError(...)
```

It keeps everything a bit more consistent shape wise?

Reply:
That's actually what I had originally, but force-pushed over. The downside to it is if someone calls get_array() it is a different shape returned, which could cause confusion?

Reply:
That is a fair point, but I am also worried about the shape stability of people who have code written against QuadMesh who are now going to be suprised that sometimes they get back 2d data.

Reply:
That should only happen if they pass in 2d data, which was not possible before. So, I think all the before cases were 1d inputs and will return 1d inputs still. This is really for my selfish future motivation of wanting to call update animations without forgetting to ravel() and get the ValueError thrown my way. I completely agree though, it does add another layer of potential confusion and that should be weighed on pros/cons.

Reply:
I am worried about the (hypothetical) that someone has written a function that takes in a `QuadMesh`, uses `get_array()`, and assumes 1d data.  If we do the reshaping at the last minute then that assumption is no longer valid, but there is no way for the function author to reasonably know.

I do see both sides of this and neither is obviously better.